apollo-router/src/test_harness.rs

```diff
@@ -89,14 +91,18 @@ impl<'a> TestHarness<'a> {
     /// Specifies the logging level. Note that this function may not be called more than once.
     /// log_level is in RUST_LOG format.
     pub fn log_level(self, log_level: &'a str) -> Self {
-        init_telemetry(log_level).expect("failed to setup logging");
+        // manually filter salsa logs because some of them run at the INFO level https://github.com/salsa-rs/salsa/issues/425
+        let log_level = format!("{log_level},salsa=error");
```

Comment:
before, salsa errors were reported under a blocking task and somehow did not appear in traces. Now that the compiler is created earlier, they appear again, so this filters them out